### PR TITLE
DVM Permissions Handling

### PR DESCRIPTION
**Pull request by pranavgaikwad**

### CODE REVIEW

File: enhancements/crane-1.x/dvm-permission-handling/README.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+Additionally, a custom UID/GID can be provided per namespace:
+
+```
+target_namespace_custom_id_mapping:
+- ns-A/<UID>:<GID>
+```
+
+If a mapping is present for a target namespace, the custom ids will be added to the Rsync Server Pod.
+
+## Alternatives
```
Comment (shawn-hurley) [inline enhancements/crane-1.x/dvm-permission-handling/README.md:142]: I would like to discuss an alternative that is out of the band of MTC entirely, for instance, a crane 2 command or a set of actions to take as a user in a knowledge base.

**Review by shawn-hurley — CHANGES_REQUESTED** [review]
I am personally worried about adding this set of code to MTC because of the way it could cause issues on regression or a bug. I would like to discuss some other alternatives.

File: enhancements/crane-1.x/dvm-permission-handling/README.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+For the controller to take any further action, the feature needs to be enabled through a _MigrationController_ flag. By default, the controller will not change file ownership and will continue to work as before i.e. by preserving file ownership and permissions. See [Upgrade/Downgrade Section](#upgrade-downgrade-strategy) for more details on the flag.
+
+If the feature is enabled, DVM will not preserve file ownership information. During the transfer, DVM will use Rsync Options and Rsync Pod's Security Context in conjuction to write the files with the correct UID in the destination cluster.
+
+### Changing ownership
+
+- The Rsync Server Pod will be forced to use the UID of the destination namespace. To enable that, we will add a new SCC which will be a clone of existing `rsync-anyuid` SCC with some changes. The changes are explained in [SCC changes section](#scc).
```
Comment (alaypatel07) [inline enhancements/crane-1.x/dvm-permission-handling/README.md:71]: how do you propose to make this change? I assume this has to be in mig-operator? do we want to add the same thing for legacy-operator or will this feature only be available in newer versions?

Comment (alaypatel07) [reply enhancements/crane-1.x/dvm-permission-handling/README.md:71]: actually nevermind, I found the answer listed below

File: enhancements/crane-1.x/dvm-permission-handling/README.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+- Rsync options `--owner`, `--group` and `--devices` will be removed from the default Rsync command options. `--owner` & `--group` options preserve UID/GID of the files. By removing these options, UID/GID of the files will not be preserved. Since Rsync Server Pod will use the UID/GID assigned to the process, the files will be written using those values. `--devices` option transfers device files, but since the Rsync Pod will not be running as _root_ anymore, that option will be removed. All other options will remain as they existed before. 
+
+- No changes will be made in security context of Rsync Client Pod (source side). It will continue to use `rsync-anyuid` SCC as it before. 
+
+#### SCC
+
+A new scc `rsync-restricted-uid` will be added to address the needs of this use-case. For the most part, this is a clone of `rsync-anyuid` SCC except following changes:
```
Comment (alaypatel07) [inline enhancements/crane-1.x/dvm-permission-handling/README.md:79]: Can you please elaborate on why a new SCC will be needed and we cant use the `restricted` SCC so future us can refer back?

**Comment by jwmatthews** [comment]
@pranavgaikwad would you take a look and see if we need to make any other updates to this so we can get it ready to merge?

**Comment by pranavgaikwad** [comment]
Closing as this is no longer considered.

**Pull request closed without merging**